gammapy/time/lomb_scargle.py

```diff
@@ -0,0 +1,237 @@
+def _freq_grid(time, dt):
+    """
+    Generates the frequency grid for the periodogram
+    """
+    max_period = (np.max(time) - np.min(time))
+    min_period = dt
+    n_periods = max_period / min_period
+    periods = np.linspace(min_period, max_period, n_periods)
+    grid = 1. / periods
+
+    return grid
+
+
+def _significance_pre(time, freq, psd_best_period):
+    """
+    Computes significance for the pre-defined beta distribution
+    """
+    from scipy.stats import beta
+    a = (3-1)/2
+    b = (len(time)-3)/2
+    significance = 100 * beta.cdf(psd_best_period, a, b)**len(freq)
+
+    return significance
+
+
+def _significance_cvm(time, freq, psd, psd_best_period):
+    """
+    Computes significance for the cvm-distance-minimised beta distribution
+    """
+    from scipy import optimize
+    from scipy.stats import beta
+    theta_1 = -1. * (np.mean(psd) * (-np.mean(psd)
+                                     + np.mean(psd)**2
+                                     + np.var(psd))
+                     ) / (np.var(psd))
+    if theta_1 < 0:
+        theta_1 = 0.00001
+    theta_2 = (theta_1 - theta_1 * np.mean(psd)) / np.mean(psd)
+    if theta_2 < 0:
+        theta_2 = 0.00001
+    cvm_minimize = optimize.fmin(_cvm, [theta_1, theta_2], args=(psd,))
+    significance = 100 * beta.cdf(psd_best_period, cvm_minimize[0], cvm_minimize[1])**len(freq)
+
+    return significance
+
+
+def _significance_nll(time, freq, psd, psd_best_period):
+    """
+    Computes significance for the negative logarithmic likelihood-minimised beta distribution
+    """
+    from scipy import optimize
+    from scipy.stats import beta
+    a = (3-1)/2
+    b = (len(time)-3)/2
+    nll_minimize = optimize.fmin(_nll, [a, b], args=(psd,))
+    significance = 100 * beta.cdf(psd_best_period, nll_minimize[0], nll_minimize[1])**len(freq)
+
+    return significance
+
+
+def _significance_boot(time, flux, flux_error, freq, psd_best_period, n_bootstraps):
+    """
+    Computes significance for the bootstrap-resampling
+    """
+    from scipy import stats
+    max_periods = _bootstrap(time, flux, flux_error, freq, n_bootstraps)
+    significance = stats.percentileofscore(max_periods, psd_best_period)
+
+    return significance
+
+
+def _significance_all(time, flux, flux_error, freq, psd, psd_best_period, n_bootstraps):
+    """
+    Computes significance for all significance criteria
+    """
+    significance = np.empty([4])
+    significance[0] = _significance_pre(time, freq, psd_best_period)
+    significance[1] = _significance_nll(time, freq, psd, psd_best_period)
+    significance[2] = _significance_cvm(time, freq, psd, psd_best_period)
+    significance[3] = _significance_boot(time, flux, flux_error, freq, psd_best_period, n_bootstraps=100)
+
+    return significance
+
+
+def lomb_scargle(time, flux, flux_error, dt, criterion='None', n_bootstraps=100):
+    """
+    This function computes the significance of periodogram peaks under certain significance criteria.
```

Comment:
Always start a function with a one-line summary, then an empty line and then more explanations.
The phrase "This function" can be removed since it doesn't contain any new info, e.g. this would be OK I think:
```
Compute period and significance of a light curve using Lomb-Scargle PSD.
```